### ISSUE TITLE
[BUG] [UI] v1.4.0-rc4 Volume page shows error

### ISSUE THREAD

**Issue by m-ildefons** [harvester/harvester#6927]
**Describe the bug**
Volume page shows errors instead of volumes

**To Reproduce**
1. Install Harvester v1.4.0-rc4
2. Install Rancher v2.9.2 via vcluster plugin
3. Run Harvester/Rancher integration tests for RKE version v1.28. These tests should pass
4. Run Harvester/Rancher integration tests for RKE version v1.27. These tests failed
5. Visit Volume page in Harvester UI

**Expected behavior**
The volumes should be displayed without error

**Support bundle**
N/A. Can't generate support bundle due to "Network Error"

**Environment**
 - Harvester ISO version: v1.4.0-rc4
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): KVM, ipxe-examples, custom config

**Additional context**
![Screenshot at 2024-11-01 12-32-43](https://github.com/user-attachments/assets/5f5450af-69ed-4a0c-a5b6-dc1fe56e6f88)

**Comment by a110605** [comment]
Hi @m-ildefons, I can't reproduce this error when importing harvester v1.4.0-rc4 to rancher v2.9.2.

Is there any specific step to trigger this error?


https://github.com/user-attachments/assets/f86f87dd-4161-4eed-90c9-08a58d0b5549

**Comment by m-ildefons** [comment]
I'm not sure how to trigger this error reliably. I encountered this error after running [tests](https://github.com/harvester/tests/) against the Rancher + Harvester instances.

Since I was running the tests in a ipxe-examples environment, it's entirely possible that something there had gone wrong and that a real bare-metal setup won't show this fault ever.

**Comment by torchiaf** [comment]
I think it's a more generic error, a user is reporting the same error in Storage Class page:

![image](https://github.com/user-attachments/assets/c8479c37-3c0d-47de-b79a-1e0de7a191cc)

I suspect it comes from the recent schemaDefinitions changes @a110605 .

**Comment by tserong** [comment]
I've figured out a reliable reproducer:

1. Create a new storage class
2. Create a volume using that storage class
3. Delete the storage class created in step 1
4. Go to the Volumes screen and you'll get the error "TypeError: this.$el.querySele ctor is not a function". I assume this is because the volume we created earlier now refers to a non-existent storage class?

I also got the same error on the Storage Classes screen when installing the democratic-csi driver. Not sure if this wants a separate issue or not, but here's the steps to reproduce:

1. Run this (ignore the fact that we're not setting proper values, we don't intend to actually _use_ freenas here, we just want to break the Harvester GUI):
    ```
    > helm repo add democratic-csi https://democratic-csi.github.io/charts/
    > helm repo update
    > helm search repo democratic-csi
    > wget https://raw.githubusercontent.com/democratic-csi/charts/refs/heads/master/stable/democratic-csi/examples/freenas-iscsi.yaml
    > helm install --values freenas-iscsi.yaml freenas-iscsi democratic-csi/democratic-csi
    ```
2. Go to the `Advanced: Storage Classes` screen and you'll get the error "TypeError: this.$el.querySelector is not a function". Must be something about the "freenas-iscsi-csi" Storage Class created when installing democratic-csi that Harvester doesn't like for some reason.

**Comment by a110605** [comment]
> I've figured out a reliable reproducer:
> 
> 1. Create a new storage class
> 2. Create a volume using that storage class
> 3. Delete the storage class created in step 1
> 4. Go to the Volumes screen and you'll get the error "TypeError: this.$el.querySele ctor is not a function". I assume this is because the volume we created earlier now refers to a non-existent storage class?
> 
> I also got the same error on the Storage Classes screen when installing the democratic-csi driver. Not sure if this wants a separate issue or not, but here's the steps to reproduce:
> 
> 1. Run this (ignore the fact that we're not setting proper values, we don't intend to actually _use_ freenas here, we just want to break the Harvester GUI):
>    ```
>    > helm repo add democratic-csi https://democratic-csi.github.io/charts/
>    > helm repo update
>    > helm search repo democratic-csi
>    > wget https://raw.githubusercontent.com/democratic-csi/charts/refs/heads/master/stable/democratic-csi/examples/freenas-iscsi.yaml
>    > helm install --values freenas-iscsi.yaml freenas-iscsi democratic-csi/democratic-csi
>    ```
> 2. Go to the `Advanced: Storage Classes` screen and you'll get the error "TypeError: this.$el.querySelector is not a function". Must be something about the "freenas-iscsi-csi" Storage Class created when installing democratic-csi that Harvester doesn't like for some reason.

Hi @tserong , @bk201 , @torchiaf , I opened https://github.com/harvester/harvester/issues/6970 to track storage class broken issue.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.4-ce5e24e3-head` (24/11/11). Close this issue.

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Volume without storage class would show type error on volumes page $~~$</summary>


1. When we delete storage class of the volume and check the volumes page 

2. The volume page can show correct content  


https://github.com/user-attachments/assets/1663825e-ef91-4d6f-8849-d003d11ab313

![image](https://github.com/user-attachments/assets/f74cd183-fbf4-4407-a803-7f788f015fd3)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Run e2e Rancher integration test and check the volume page status $~~$</summary>

1. After complete the e2e test for rancher integration 
2. Access the volume page can display correctly without error 

  https://github.com/user-attachments/assets/9e1afdbf-8a32-4880-b35a-4676dd1df99a


 </details>

### Test Information
* Test Environment: Three nodes harvester on kvm machines
* Harvester version: `v1.4-ce5e24e3-head` (24/11/11)
* Rancher version: `v2.9.3` 

### Verify Steps
<details><summary> Volume without storage class would show type error on volumes page </summary>

1. Create a new storage class named `sc3` on node-2
  ![image](https://github.com/user-attachments/assets/8c8760f3-1992-4085-80f7-ae26ca14d7bd)

3. Create a volume using that storage class name `volume-sc3`

  ![image](https://github.com/user-attachments/assets/19f18f69-aeab-4cc7-855f-9b537729919b)


4. Delete the storage class `sc3`created in step 1

5. Open the Volumes page and check no error message prompt


</details>

<details><summary> Run e2e Rancher integration test and check the volume page  </summary>

1. Execute Rancher integration e2e test suite 

  ```
  tox -e py36 --result-json=test_result.json -- -l harvester_e2e_tests/integrations --json-report --html=test_report/test_result.html -m 'not delete_host and not upgrade' -k 'test_9_rancher_integration'
  ```

2. After e2e test complete, open the volume page and check the content 

 

</details>
